samples/subsys/power/device_pm/src/dummy_driver.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <misc/printk.h>
+#include "dummy_parent.h"
+#include "dummy_driver.h"
+
+struct k_poll_signal async_sig = K_POLL_SIGNAL_INITIALIZER(async_sig);
+struct k_poll_event  async_evt =
+		K_POLL_EVENT_INITIALIZER(K_POLL_TYPE_SIGNAL,
+					 K_POLL_MODE_NOTIFY_ONLY,
+					 &async_sig);
+u32_t device_power_state;
+
+static struct device *parent;
+
+static int dummy_open(struct device *dev)
+{
+	int ret, result;
+	u32_t signaled;
+
+	printk("open()\n");
+	ret = device_pm_get(dev);
+	if (ret != 1) {
+		return ret;
+	}
+	printk("async wakeup request queued\n");
+	ret = k_poll(&async_evt, 1, K_FOREVER);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (async_evt.state == K_POLL_STATE_SIGNALED) {
+		k_poll_signal_check(&async_sig, &signaled, &result);
+		if (result == DEVICE_PM_STATUS_ACTIVE) {
+			printk("Dummy device resumed\n");
+			ret = 0;
+		} else {
+			printk("Dummy device Not resumed\n");
```

Comment:
IMHO we have to wait here in a loop, as device might go through SUSPENDED state if put() was executed just before requesting get().

Reply:
which block should go in loop?

Reply:
We might get device in proper state in 2nd attempt. So whole waiting for event should be in loop in my opinion.